cloudimg/ms_azure.py

```diff
@@ -354,6 +363,41 @@ def upload_callback(self, response):
 
         return self._upload_progress
 
+    def get_blob_copy_status(self, blob_client):
+        """
+        Wait  and confirm till the copy of blob from url is completed.
+
+        Args:
+            blob_client : Blob client which initiated the copy process.
+        Returns:
+            Upload blob client on success
+        """
+        get_copy_progess = (
+            lambda uploaded, total: (int(uploaded) / int(total))*100
+        )
+        try:
+            copy_status = (
+                lambda blob_properties:
+                blob_properties["copy"]["status"] == "success"
+            )
+            for attempt in Retrying(
+                stop=stop_after_attempt(60) | stop_after_delay(600),
+                retry=retry_if_not_result(copy_status),
+                wait=wait_fixed(10),
+            ):
+                with attempt:
+                    blob_properties = blob_client.get_blob_properties()
+                attempt.retry_state.set_result(blob_properties)
+                copy_status = blob_properties["copy"]["status"]
+                copy_progress = round(
+                    get_copy_progess(
+                        *blob_properties["copy"]["progress"].split("/")), 2
+                )
+                log.info(f"Copying in progress : {copy_progress} %")
+            return blob_client
+        except RetryError:
+            log.error("Unable to confirm if the blob was copied successfully.")
```

Comment:
The behavior on error doesn't seem right.

This method is called by `upload_to_container` which is documented as:

```
        Returns:
            The uploaded BlobClient on success.
        Raises:
            AzureError on upload failure.
```

However, what you've implemented here will result in `upload_to_container` simply returning `None` instead if the copy doesn't complete.

It seems like you should ensure that some kind of exception is raised if the copy doesn't complete?  Otherwise it's almost certain that somebody at some point won't check the result and will write code which incorrectly succeeds even when the copy failed.

Also, the error log here is rather vague, is there any more detail which could be included?  Perhaps there could be something useful to log in `blob_properties["copy"]` ?